src/check/canonicalize.zig

```diff
@@ -26,8 +26,14 @@ var_function_regions: std.AutoHashMapUnmanaged(CIR.Pattern.Idx, Region),
 var_patterns: std.AutoHashMapUnmanaged(CIR.Pattern.Idx, void),
 /// Tracks which pattern indices have been used/referenced
 used_patterns: std.AutoHashMapUnmanaged(CIR.Pattern.Idx, void),
-/// Keep track of pending annotation
+/// Scratch type variables
 scratch_vars: std.ArrayListUnmanaged(TypeVar),
+/// Scratch ident
+scratch_idents: std.ArrayListUnmanaged(Ident.Idx),
+/// Scratch ident
+scratch_record_fields: std.ArrayListUnmanaged(types.RecordField),
+/// Scratch ident
+scratch_seen_record_fields: std.ArrayListUnmanaged(SeenRecordField),
```

Comment:
Should we be using the common Scratch implementation at `src/base/Scratch.zig`?

Reply:
Yup, great idea!